### ISSUE TITLE
Is it possible to change the display of `project.rs.xml` in the open dialog to the name of the project?

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/QRAVEPlugin#55]
@MattReimer please don't read or respond to this now... I am queuing up for when we finally sit down and prioritize warehouse development and make some decisions.

## The Problem
For experienced users, we no longer think twice about the fact that every project (no matter how different) has the exact same name of `project.rs.xml`.  However, this is just wrong. It is unintuitive to new and existing users. 

![image](https://user-images.githubusercontent.com/9723905/135162935-d130fcdc-9b1f-43ed-959d-b63be2bb7c78.png)

## The Request
Can we trick the display of the project name on open dialog to NOT be reading actual `project.rs.xml` file name on disk and instead read the `<Name>` as shown above? I'm suggesting this because I am guessing @MattReimer will not like idea of renaming `project.rs.xml` files in warehouse? I appreciate that you are likely using a bog-standard open file dialog/control, and this is not possible without making a custom interface for navigating disk to look for those. 

## The Acknowledgment
This has bugged me for a long time but I have never said anything about it (sorry). I like what we did in GCD (i.e. make a safe project name based on removing spaces and odd characters from equivalent of what's in <Name> and add our own custom extension (e.g. if Project `<Name>` was "Glenn Feshie" the file name would be `GlennFeshie.gcd` , even though `*.gcd` is really just an xml. That makes way more sense. The extension of that logic would be to have a file extension for every project type (e.g. for VBET maybe `*.rs.vbet`, for BRAT `*.rs.brat`, etc.). Maybe that is straightforward? Maybe that is a nightmare. However, I have had multiple users mention how this initially confused them.

**Comment by MattReimer** [comment]
I agree 100% that `project.rs.xml` files aren't intuitive but "Do away with file browser dialogs!" is not the way to go. In QGIS you can use the catalog, you can open a shp file manually OR you can drag a file on top of the app. Seems to me if we want our own project catalog feature we should implement it on top of what we already have and put a catalog icon into our toolbar.

I'm not the end user here, granted, but I'm trying to think about how people use their machines. I really don't want to lose the ability to quickly load a project from a different location.

For example: I might keep the major stash of projects on a USB drive I plug in because I don't have enough room on my laptop (based on a true story). Then I might also have a smaller stash of projects I use often on my small SSD hard drive. 

Right now I can open projects wherever they are: box, usb , downloads folder, whatever. Could be really annoying to have to change the root folder every time I want to load a project from a different area.

### Minor change:

If we do implement this I'd also add an extra folder level if I were you. The case I'm thinking of is that I have two identical projects that ran a day apart side by side. If you just use the project name you lose the context of the folder name you chose and thus, the ability to tell them apart. Also project names have no guarantee to identify a project in any kind of unique way.

so if the projects look like this on the file system:

```
/catalogdir/BRAT/17060304BRAT_yesterday_run/project.rs.xml
/catalogdir/BRAT/17060304BRAT_today_run/project.rs.xml
```

They should look like this in the catalog tree:

```
-ROOT(folder)
    -BRAT(folder)
        -17060304BRAT_yesterday_run(folder)
                -BRAT Run for HUC 17060304    <-------This is what I can select and load
        -17060304BRAT_today_run(folder)
                -BRAT Run for HUC 17060304    <-------This is what I can select and load
```

### In summary:

1. Nothing disqualifying here so I'm definitely not saying we shouldn't implement a project catalog feature
2. but maybe we could load projects from disk too?
3. We need to think carefully about edge cases, as usual. This kind of problem always gets hairy on the edges.
    - eventually we may end up with a nest-y folder with thousands of projects. Can we scrape this fast enough that the UI doesn't become unusable?
    - Think carefully about things like duplicate projects and the messy and different ways people organize their hard drives.
    - Symlinks? Mounted drives? Cross platform concerns? Network drives? Slow connections? disconnected drives? File locking on windows?

**Comment by joewheaton** [comment]
This is still an important issue to me.

**Comment by MattReimer** [comment]
Now that QRiS is a thing in the world we should include it in this conversation.... when we have it.
